cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java

```diff
@@ -587,70 +389,62 @@ public void testReconcileUpdatesKafkaStorageType(Params params, VertxTestContext
+                    context.failNow(new Exception("If storage is not persistent, something has gone wrong"));
                 }
-                kafkaAssembly(NAMESPACE, CLUSTER_NAME).patch(PatchContext.of(PatchType.JSON), updatedStorageKafka);
+                kafkaAssembly(NAMESPACE, CLUSTER_NAME).patch(updatedStorageKafka);
 
                 LOGGER.info("Updating with changed storage type");
             })))
             .compose(v -> operator.reconcile(new Reconciliation("test-trigger", Kafka.RESOURCE_KIND, NAMESPACE, CLUSTER_NAME)))
             .onComplete(context.succeeding(v -> context.verify(() -> {
                 // Check the Volumes and PVCs were not changed
-                assertPVCs(context, KafkaResources.kafkaStatefulSetName(CLUSTER_NAME), originalPVCs.get());
-                assertVolumes(context, KafkaResources.kafkaStatefulSetName(CLUSTER_NAME), originalVolumes.get());
-                assertInitContainers(context, KafkaResources.kafkaStatefulSetName(CLUSTER_NAME), originalInitContainers.get());
+                assertPVCs(context, CLUSTER_NAME + "-kafka");
+                assertVolumes(context, CLUSTER_NAME + "-kafka");
                 async.flag();
             })));
     }
 
-    private void assertPVCs(VertxTestContext context, String statefulSetName, List<PersistentVolumeClaim> originalPVCs) {
-        context.verify(() -> {
-            StatefulSet statefulSet = client.apps().statefulSets().inNamespace(NAMESPACE).withName(statefulSetName).get();
-            assertThat(statefulSet, is(notNullValue()));
-            List<PersistentVolumeClaim> pvcs = statefulSet.getSpec().getVolumeClaimTemplates();
-            assertThat(originalPVCs.size(), is(pvcs.size()));
-            assertThat(originalPVCs, is(pvcs));
-        });
 
-    }
-
-    private void assertVolumes(VertxTestContext context, String statefulSetName, List<Volume> originalVolumes) {
+    private void assertPVCs(VertxTestContext context, String podSetName) {
         context.verify(() -> {
-            StatefulSet statefulSet = client.apps().statefulSets().inNamespace(NAMESPACE).withName(statefulSetName).get();
-            assertThat(statefulSet, is(notNullValue()));
-            List<Volume> volumes = statefulSet.getSpec().getTemplate().getSpec().getVolumes();
-            assertThat(originalVolumes.size(), is(volumes.size()));
-            assertThat(originalVolumes, is(volumes));
+            assertThat(kafkaStorage.getType(), is("persistent-claim"));
+            List<PersistentVolumeClaim> pvc = new ArrayList<>();
+            client.persistentVolumeClaims().inNamespace(NAMESPACE).list().getItems().stream().forEach(persistentVolumeClaim -> {
+                if (persistentVolumeClaim.getMetadata().getName().contains(podSetName)){
+                    pvc.add(persistentVolumeClaim);
+                }
+            });
+            assertThat(pvc.size(), is(3));
```

Comment:
Aren't there more properties to check? Storage class etc.?

Reply:
I have added the assertion for storage class

Reply:
Maybe you can check also for example size? The `println` can be maybe removed and I think you have some checkstyle errors.